microceph/api/s3.go

```diff
@@ -0,0 +1,79 @@
+package api
+
+import (
+	"encoding/json"
+	"net/http"
+
+	"github.com/canonical/lxd/lxd/response"
+	"github.com/canonical/microceph/microceph/api/types"
+	"github.com/canonical/microceph/microceph/ceph"
+	"github.com/canonical/microcluster/rest"
+	"github.com/canonical/microcluster/state"
+)
+
+// /1.0/resources endpoint.
+var s3Cmd = rest.Endpoint{
+	Path:   "services/rgw/user",
+	Get:    rest.EndpointAction{Handler: cmdS3Get, ProxyTarget: true},
+	Put:    rest.EndpointAction{Handler: cmdS3Put, ProxyTarget: true},
+	Delete: rest.EndpointAction{Handler: cmdS3Delete, ProxyTarget: true},
+}
```

Comment:
For optimal RESTy-ness I'd like to suggest an API structure like this:

GET /services/rgw/user  # returns a list of users
GET /services/rgw/user/{uid} # returns info about a specific user

POST /services/rgw/user # create a new user 

PUT /services/rgw/user/{name} # update existing user

DELETE /services/rgw/user/{name} # delete user

Something like this can be used to specify a path with a {uid} var:

```
var s3Cmd = rest.Endpoint{
	Path:   "services/rgw/user/{name}",
	Get:    rest.EndpointAction{Handler: cmdS3GetSingle, ProxyTarget: true},
        Put: 
...

func cmdS3GetSingle(s *state.State, r *http.Request) response.Response {
	var err error
	var name string
	name, err = url.PathUnescape(mux.Vars(r)["name"])
	if err != nil {
		return response.BadRequest(err)
	}
	// do something with user name
	user = ceph.GetS3User(name)
...
```